src/compile/repeat.ts

```diff
@@ -0,0 +1,224 @@
+import {Channel} from '../channel';
+import {CellConfig, Config} from '../config';
+import {Encoding} from '../encoding';
+import {Facet} from '../facet';
+import {Field, FieldDef, isRepeatRef} from '../fielddef';
+import * as log from '../log';
+import {Repeat} from '../repeat';
+import {RepeatSpec} from '../spec';
+import {Dict, duplicate, vals} from '../util';
+import {VgData, VgLayout, VgScale} from '../vega.schema';
+import {buildModel} from './common';
+import {assembleData} from './data/assemble';
+import {assembleLayoutData} from './layout';
+import {Model} from './model';
+
+
+export type RepeatValues = {
+  row?: string,
+  column?: string
+};
+
+export function facetRepeatResolve(facet: Facet<Field>, repeatValues: RepeatValues): Facet<string> {
+  return resolveRepeat(facet, repeatValues);
+}
+
+export function encodingRepeatResolve(encoding: Encoding<Field>, repeatValues: RepeatValues): Encoding<string> {
+  return resolveRepeat(encoding, repeatValues);
+}
+
+type EncodingOrFacet<F> = Encoding<F> | Facet<F>;
+
+function resolveRepeat(encoding: EncodingOrFacet<Field>, repeatValues: RepeatValues): EncodingOrFacet<string> {
+  const out: EncodingOrFacet<string> = {};
+  for (const channel in encoding) {
+    if (encoding.hasOwnProperty(channel)) {
+      const fieldDef: FieldDef<Field> = encoding[channel];
+      out[channel] = {...fieldDef};
+
+      const field = fieldDef.field;
+      if (isRepeatRef(field)) {
+        if (field.repeat in repeatValues) {
+          out[channel].field = repeatValues[field.repeat];
+        } else {
+          log.warn(log.message.noSuchRepeatedValue(field.repeat));
+          delete out[channel].field;
```

Comment:
Not necessarily. What if you use count aggregate for example. But probably. I'll change it.